src/Parser/Mage/Frost/Modules/Features/BrainFreeze.js

```diff
@@ -11,81 +11,123 @@ const debug = false;
 // Brain Freeze appears to always fall after Flurry cast, but not always on same timestamp. Giving a margin here.
 const PROC_WINDOW_MS = 100;
 
+// When Glacial Spike is talented, it's OK to overwrite Brain Freeze procs if you have at least this many icicles.
+const ICICLE_BREAKPOINT = 3;
+
 class BrainFreeze extends Analyzer {
-	lastFlurryTimestamp;
+  lastFlurryTimestamp;
+
+  overwrittenProcs = 0;
+  okOverwrittenProcs = 0;
+  expiredProcs = 0;
+  totalProcs = 0;
+  flurryWithoutProc = 0;
+  currentIcicleCount = 0;
+
+  handleIcicleStacks(event) {
+    if (event.type === 'removebuff' || isNaN(event.stack)) {
+      event.stack = 0;
+    }
+    if (event.type === 'applybuff') {
+      event.stack = 1;
+    }
+    this.currentIcicleCount = event.stack;
+  }
 
-	overwrittenProcs = 0;
-	expiredProcs = 0;
-	totalProcs = 0;
-	flurryWithoutProc = 0;
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.ICICLES_BUFF.id) {
+      this.handleIcicleStacks(event);
+      return;
+    }
+    if (spellId === SPELLS.BRAIN_FREEZE.id) {
+      this.totalProcs += 1;
+    }
+  }
 
-	on_byPlayer_applybuff(event) {
-		const spellId = event.ability.guid;
-    if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
-		this.totalProcs += 1;
-	}
+  on_byPlayer_applybuffstack(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.ICICLES_BUFF.id) {
+      this.handleIcicleStacks(event);
+    }
+  }
 
-	on_byPlayer_refreshbuff(event) {
-		const spellId = event.ability.guid;
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.ICICLES_BUFF.id) {
+      this.handleIcicleStacks(event);
+    }
     if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
-		this.overwrittenProcs += 1;
-		this.totalProcs += 1;
-    debug && console.log("Brain Freeze proc overwritten @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      return;
+    }
+    this.totalProcs += 1;
+
+    if (!this.selectedCombatant.hasTalent(SPELLS.GLACIAL_SPIKE_TALENT.id)) {
```

Comment:
This one doesnt matter too terribly much, but i believe the standard is to set a variable within the constructor at the top of the file (Check out GlacialSpike, it should have this in it

constructor(...args) {
    super(...args);
    this.active = this.selectedCombatant.hasTalent(SPELLS.GLACIAL_SPIKE_TALENT.id);
    this.hasEbonbolt = this.selectedCombatant.hasTalent(SPELLS.EBONBOLT_TALENT.id);
  }

this.active doesnt need to be included, but you could do
this.hasGlacialSpike = this.selectedCombatant.hasTalent(SPELLS.GLACIAL_SPIKE_TALENT.id);

and then below do !this.hasGlacialSpike ... to shorten things a little. 

But again, its really minor and probably doesnt matter too terribly much

Reply:
I was following what was already in the file, in your opinion should I change it?